docs/source/methods/random_numbers.rst

```diff
@@ -62,11 +66,131 @@ Note that equation :eq:`lcg-skipahead` has the same general form as equation :eq
 the idea is to determine the new multiplicative and additive constants in
 :math:`O(\log_2 N)` operations.
 
+
+--------------------------------
+Permuted Congruential Generators
+--------------------------------
+
+A permuted congruential generator (PCG) aims to improve statistical quality 
+of a LCG by using permutation functions. The general algorithm consists of 
+two steps:
+
+1. advance the LCG generator according to :eq:`pcg_lcg`,
+2. output the LCG state "scrambled" by permutation function.
+
+
+
+OpenMC uses the PCG-RXS-M-XS variant with 64-bit state and 
+64-bit output. The code for permutation functiom is taken 
+from `PCG GitHub`_. The exact algorithm follows.
+
+1. Generate LCG output :math:`\xi_{i+1}`
+
+  .. math::
+      :label: pcg_lcg
+  
+      \xi_{i+1} = g \xi_i + c
+
+  where :math:`g=6364136223846793005` and :math:`c=1442695040888963407`.
+
+2. Apply random xorshift. 
+    * First, take the upper 5 bits of :math:`\xi_{i+1}` 
+      using the `arithmetic right shift operator`_ by 59 positions.
+    * Second, add :math:`5` and shift seed by that number of bits. The 
+      name "random" refers to this step, where the length of the shift 
+      is evaluated from the seed, not given as constant.
+    * Third, perform `bitwise XOR`_ with the upper bits and original :math:`\xi_{i+1}`.
+
+#. Multiply by :math:`6364136223846793005`.
+#. Apply xorshift - similarly to step 2, use `arithmetic right shift operator`_ 
+   to take upper 21 bits and perform `bitwise XOR`_. The length of 
+   the shift is constant here.
+#. Convert to double from interval :math:`[0, 1)`.
+
+
+**Advantages of PCG over LCG include**
+
+* increased statistical quality - measured by statistical tests from BigCrush library,
+* small performance burden compared to LCG.
+
+For elaborated description, see [ONeill]_.
+
+
+Example of PCG algorithm
+------------------------
```

Comment:
I don't know how to explain it in an "understandable" way, so I added an example.